### PR TITLE
Fix/ln dlc/renew after disconnect

### PR DESCRIPTION
**Pull request by Tibo-lg**
Fix for the bug mentioned [here](https://github.com/p2pderivatives/rust-dlc/pull/142#issuecomment-1695055003) where settling a dlc channel after disconnecting during the opening errors.

### CODE REVIEW

**Review by luckysori — APPROVED** [review]
LGTM!

**Comment by Tibo-lg** [comment]
@luckysori I think this is finally ready to be reviewed. There was actually several bugs related to revocation secrets. I added checks that helped find the issues, but I also opened #146 as I think it'd be better to have tests for that.

**Review by luckysori — APPROVED** [review]
LGTM!

I think the second patch is already in the `feature/ln-dlc-channels`.

File: dlc-manager/src/channel/signed_channel.rs [hunk 1]
```diff
@@ -264,12 +264,9 @@ typed_enum!(
...
             contract_id: ContractId,
-            /// The per update point to be used by the offer party for the setup
-            /// of the next channel state.
-            offer_per_update_point: PublicKey,
-            /// The per update point to be used by the accept party for the setup
-            /// of the next channel state.
-            accept_per_update_point: PublicKey,
+            /// The previous per update point that was used by the offer party for the previous
+            /// state of the channel.
+            prev_offer_per_update_point: PublicKey,
```
[the remark this answers is not in the document]
Comment (Tibo-lg) [reply dlc-manager/src/channel/signed_channel.rs:269]: IIRC the `offer_per_updated_point` was not really used and/or redundant with the `own_`/`counter_per_update_point` of the `SignedChannel` struct. The reason for caring about the previous one is to be able to verify that the revocation secret we receive later matches with the point we had.

File: dlc-manager/src/channel/ser.rs [hunk 2]
```diff
@@ -60,7 +60,7 @@ impl_dlc_writeable_enum!(
...
-    (10, RenewFinalized, {(contract_id, writeable), (offer_per_update_point, writeable), (accept_per_update_point, writeable), (buffer_transaction, writeable), (buffer_script_pubkey, writeable), (offer_buffer_adaptor_signature, {cb_writeable, write_ecdsa_adaptor_signature, read_ecdsa_adaptor_signature}), (accept_buffer_adaptor_signature, {cb_writeable, write_ecdsa_adaptor_signature, read_ecdsa_adaptor_signature}), (timeout, writeable), (own_payout, writeable), (total_collateral, writeable)}),
+    (10, RenewFinalized, {(contract_id, writeable), (prev_offer_per_update_point, writeable), (buffer_transaction, writeable), (buffer_script_pubkey, writeable), (offer_buffer_adaptor_signature, {cb_writeable, write_ecdsa_adaptor_signature, read_ecdsa_adaptor_signature}), (accept_buffer_adaptor_signature, {cb_writeable, write_ecdsa_adaptor_signature, read_ecdsa_adaptor_signature}), (timeout, writeable), (own_payout, writeable), (total_collateral, writeable)}),
```
[the remark this answers is not in the document]
Comment (holzeis) [reply dlc-manager/src/channel/ser.rs:63]: you mean any channel that is already stuck?

[the remark this answers is not in the document]
Comment (luckysori) [reply dlc-manager/src/channel/ser.rs:63]: Yeah, if a channel is currently in `RenewFinalized` we won't be able to load it after we update to this patch.

**Pull request merged**